app/controllers/devise_token_auth/registrations_controller.rb

```diff
@@ -142,11 +131,8 @@ def account_update_params
     private
 
     def resource_update_method
-      if DeviseTokenAuth.check_current_password_before_update == :attributes
-        "update_with_password"
-      elsif DeviseTokenAuth.check_current_password_before_update == :password and account_update_params.has_key?(:password)
-        "update_with_password"
-      elsif account_update_params.has_key?(:current_password)
+      if DeviseTokenAuth.check_current_password_before_update == :attributes ||
+         account_update_params.has_key?(:current_password)
```

Comment:
This was logically redundant
```
elsif DeviseTokenAuth.check_current_password_before_update == :password and account_update_params.has_key?(:password)
  "update_with_password
```
since it's automatically covered by
```
elsif account_update_params.has_key?(:current_password)
  account_update_params.has_key?(:current_password)
```

Combined the clauses into one for brevity.